pkg/roachprod/prometheus/prometheus.go

```diff
@@ -354,31 +353,36 @@ docker run --privileged -p 9090:9090 \
 
 	return c.Get(
 		l,
+		promNode,
 		"/tmp/prometheus/prometheus-snapshot.tar.gz",
 		dir,
 	)
 }
 
-func StopPrometheus(
+func Shutdown(
 	ctx context.Context, c *install.SyncedCluster, l *logger.Logger, nodes install.Nodes,
 ) error {
 
+	// We currently assume the last node contains the server.
+	promNode := install.Nodes{nodes[len(nodes)-1]}
```

Comment:
Is this buggy? What happens if you `roachprod stop-grafana $CLUSTER:2` when grafana is running on $CLUSTER:4? The thing is in all this roachprod code, `<cluster>` is overloaded in that it allows for a node selector. Which is why to get to exactly the node in question, we want something like:

```
nodes, err := install.ListNodes("all", len(c.VMs))
if err != nil {
  return err
}
promNode := nodes[len(nodes)-1]
```

This PR has this kind of bug in the other place it's trying to use `TargetNodes`.

Reply:
Done.